src/utils/utils.cpp

```diff
@@ -255,3 +266,29 @@ DEVICE_DIRECTION Utils::getIioDeviceDirection(struct iio_device* dev)
 	}
 	return dir;
 }
+
+int Utils::compareVersions(std::string v1, std::string v2)
+{
+	while ((v1.find(".") != std::string::npos) ||
+			(v2.find(".") != std::string::npos)) {
+		if (v1.find(".") != std::string::npos) {
+			v1.erase(v1.find("."), 1);
+		}
+		if (v2.find(".") != std::string::npos) {
+			v2.erase(v2.find("."), 1);
+		}
+	}
+	if (v1.size() < v2.size()) {
+		std::string surplus("0", (v2.size() - v1.size()));
+		v1.append(surplus);
+	} else if (v2.size() < v1.size()) {
+		std::string surplus("0", (v1.size() - v2.size()));
+		v2.append(surplus);
+	}
+	if (v1 == v2) {
+		return 0;
+	} else if (v1 < v2) {
+		return 1;
+	}
+	return -1;
```

Comment:
A more compact for of this can be this one liner:
`return v1.compare(v2);`

Reply:
Updated.